### PR TITLE
Small crosscompiling issues

### PR DESCRIPTION
**Pull request by sgh**
Hi

Please merge my changes - or maybe comment on any issues I have missed noticing.

Thanks.

### CODE REVIEW

**Comment by sourcefrog** [comment]
Don't worry about the Appveyor build, it's not working yet.

Rather than renaming `buf.h`, I'd rather just move the contents into `librsync.h` and have that as the only public header.  I'd rather if you did one issue per PR.  If it's going to be public it should have comments, which might be able to just move from the `.c` file.

**Comment by sgh** [comment]
Thanks for the comments. I do not think that the move of the declaration is necessary after all. My usecase involved storing and loading deltas, signatures and patches from in-memory buffers. I recently found out that fmemopen can solve that problem for me.

Do you agree with that - or do you still think that the buf.h declarations should be moved to librsync.h? In case it should be moved I'll happily fix the things you mention and issue new pull requests.

**Comment by sourcefrog** [comment]
You should be able to do that using the existing buffer management in
librsync.h.

**Comment by sourcefrog** [comment]
Please put up a new pr skipping detection of Perl when we don't intend to run the tests on the host machine. 

Let's discuss separately in an issue how you should handle memory buffers.

File: CMakeLists.txt [hunk 1]
```diff
@@ -306,14 +306,13 @@ if (BUILD_RDIFF)
       src/rdiff.c
       src/isprefix.c)
 
-  add_executable(rdiff ${rdiff_SRCS})
   if (POPT_FOUND)
+    add_executable(rdiff ${rdiff_SRCS})
```
Comment (gulikoza) [inline CMakeLists.txt]: Isn't this already handled with BUILD_RDIFF and set(LAST_TARGET rsync) ?

**Comment by sgh** [comment]
It turns out that fmemopen is not available on windows (which is a platform we target as well as linux). So possibly a windows-way of doing it is to create tmpfiles for it.

Regarding rdif it might be that rdiff can be turned off explicitly. I do however think that autodetecting stuff is beneficial.

I'll issue new pr's for each of the topics and take it from there. So I'll close the pr. Thank you for helping me.

**Pull request closed without merging**